File: 2020/6/6.py
inp = open("input.txt").read().split("\n\n")
s = s2 = 0
for i in inp:
    p = i.split("\n")
    q = [*"".join(p)]
    s += len(set(q))
    s2 += sum(1 if q.count(x) == len(p) else 0 for x in set(q))

# part a, b
print(s, s2)

File: 2021/06/6.rb
def solve c, mature, newborns
  c.times do
    mp = mature[0]
    nbp = newborns[0]
    mature = mature.rotate
    mature[6] += nbp
    newborns = newborns.rotate
    newborns[8] = mp + nbp
  end
  mature.sum + newborns.sum
end

mature = Array.new(7,0)
newborns = Array.new(9,0)
open("input.txt").read.split(?,).map(&:to_i).each{ mature[_1] += 1 }

p solve  80, mature, newborns
p solve 256, mature, newborns
